lib/starknet_explorer_web/live/pages/block_detail.ex

```diff
@@ -8,8 +9,23 @@ defmodule StarknetExplorerWeb.BlockDetailLive do
 
   defp get_block_proof(block_hash) do
     try do
-      response = S3.get_object!("#{block_hash}" <> "-proof")
-      :erlang.binary_to_list(response.body)
+      case Application.get_env(:starknet_explorer, :prover_storage) do
+        "s3" ->
+          response = S3.get_object!("#{block_hash}" <> "-proof")
+          :erlang.binary_to_list(response.body)
+
+        _ ->
+          proofs_dir = Application.get_env(:starknet_explorer, :proofs_root_dir)
+
+          case File.read(Path.join(proofs_dir, "#{block_hash}" <> "-proof")) do
```

Comment:
This Path.join will fail if proofs_dir is nil, although we won't notice it because of the try/do clause.
Again, do we want this to fail silently or do we want to notice it?

Reply:
It is ok if it fails and returns `:not_found`, but I set a default value for the env var since it probably is better to have one anyway